lib/export/torg_mail_ru_exporter.rb

```diff
@@ -1,111 +1,73 @@
-# -*- coding: utf-8 -*-
 module Export
-  class TorgMailRuExporter
-    include Rails.application.routes.url_helpers
-    attr_accessor :host, :currencies
-    
-    SCHEME = Nokogiri::XML('<!DOCTYPE torg_price SYSTEM "shops.dtd" />')
-    MCP = 3 # Максимальная цена клика в рублях
+  class TorgMailRuExporter < YandexMarketExporter
+    # http://torg.mail.ru/info/122/
+    # cbid - ставка для управления позицией товара в карточке модели. Задается в российских рублях.
+    CBID = '10'
 
-    def helper
-      @helper ||= ApplicationController.helpers
-    end
-    
     def export
-      # @config = ::TorgMailRu.find_or_create_by_name('Default configuration')
-      @config = ::YandexMarket.first
+      @config = Spree::YandexMarket::Config.instance
       @host = @config.preferred_url.sub(%r[^http://],'').sub(%r[/$], '')
       ActionController::Base.asset_host = @config.preferred_url
-      
+
       @currencies = @config.preferred_currency.split(';').map{|x| x.split(':')}
       @currencies.first[1] = 1
-      
-      @categories = Taxon.find_by_name(@config.preferred_category)
-      @categories = @categories.self_and_descendants
-      @categories_ids = @categories.collect { |x| x.id }
-      
-      Nokogiri::XML::Builder.new({ :encoding =>"utf-8"}, SCHEME) do |xml|
-        xml.torg_price(:date => Time.now.to_s(:ym)) {
-          
-          xml.shop { # описание магазина
-            xml.shopname    @config.preferred_short_name
+
+      Nokogiri::XML::Builder.new(encoding: 'utf-8') do |xml|
+        xml.torg_price(date: Time.now.to_s(:ym)) do
+
+          xml.shop do
+            xml.name    @config.preferred_short_name
             xml.company @config.preferred_full_name
             xml.url     path_to_url('')
-            
-            xml.currencies { # описание используемых валют в магазине
-              @currencies && @currencies.each do |curr|
-                opt = {:id => curr.first, :rate => curr[1] }
-                opt.merge!({ :plus => curr[2]}) if curr[2] && ["CBRF","NBU","NBK","CB"].include?(curr[1])
-                xml.currency(opt)
+
+            xml.currencies do
+              xml.currency id: 'RUR', rate: '1'
+            end
+
+            xml.categories do
+              Spree::Taxon.published.where(export_to_yandex_market: true).drop(1).each do |taxon|
```

Comment:
If `drop(1)` is intended for skipping the `root` taxon, than I suggest to use a more explicit solution, something like this: 
```ruby
Spree::Taxon.published.where(export_to_yandex_market: true).except(Spree::Taxon.root)
# or
Spree::Taxon.published.where(export_to_yandex_market: true) - [Spree::Taxon.root]
```